src/model/Document.cpp

```diff
@@ -407,13 +407,6 @@ auto Document::operator=(const Document& doc) -> Document& {
     buildContentsModel();
     updateIndexPageNumbers();
 
-    bool lastLock = tryLock();
-    unlock();
```

Comment:
Here we unlock in any of the two cases.

Reply:
It's worst than that. From Gtk doc: 
> Calling `g_mutex_unlock()` on a mutex that is not locked by the current thread leads to undefined behaviour.

So if the previous `trylock` returned false (i.e. the mutex is already locked, we don't know by whom), we're in uncharted territories.